src/platform/unix/mod.rs

```diff
@@ -500,6 +501,8 @@ impl OsIpcReceiverSet {
                         }
                         Err(err) if err.channel_is_closed() => {
                             self.pollfds.remove(&evt_token).unwrap();
+                            let io = EventedFd(&poll_entry.fd);
+                            self.poll.deregister(&io).unwrap();
```

Comment:
> This makes total sense now

I know right. After I saw this, it seemed plain as day.

> Explicitly removing the FD should indeed fix this for good. Have you tried whether this alone fixes the failures?

Yes. Just removing the fd causes the tests to pass.

> BTW, is there any particular reason why you introduced the io temporary?

`gdb` if I remember right. Had a breakpoint that printed its value. I can remove it.